cypress/integration/change_password.spec.js

```diff
@@ -106,67 +106,63 @@ describe('Password change flow', () => {
     it('displays a breached error', () => {
       cy.mockNext(200);
-      page.goto(fakeToken);
-      page.submitPasswordChange('password', 'password');
-      cy.contains(ChangePasswordPage.CONTENT.ERRORS.PASSWORD_BREACHED);
+      cy.visit(`/reset-password/fake_token`);
+      cy.get('input[name="password"]').type('password');
+      // there seems to be a race condition on the change password page
+      // adding a small wait fixes the issue
+      cy.wait(300);
+      cy.get('button[type="submit"]').click();
+      cy.contains('common password');
     });
   });
 
   context('CSRF token error on submission', () => {
     it('should fail on submission due to CSRF token failure if CSRF token cookie is not sent', () => {
       cy.mockNext(200);
-      page.goto(fakeToken);
+      cy.visit(`/reset-password/fake_token`);
       cy.clearCookie('_csrf');
-      page.submitPasswordChange(
-        'thisisalongandunbreachedpassword',
-        'thisisalongandunbreachedpassword',
-      );
-      cy.contains(ChangePasswordPage.CONTENT.ERRORS.CSRF);
+      cy.get('input[name="password"]').type('thisisalongandunbreachedpassword');
+      // there seems to be a race condition on the change password page
+      // adding a small wait fixes the issue
+      cy.wait(300);
+      cy.get('button[type="submit"]').click();
+      cy.contains('Please try again.');
     });
   });
 
   context('Enter and Confirm passwords left blank', () => {
     it('uses the standard HTML5 empty field validation', () => {
       cy.mockNext(200);
-      page.goto(fakeToken);
-      page.clickPasswordChange();
-      page.invalidPasswordChangeField().should('have.length', 1);
-      page.invalidPasswordChangeConfirmField().should('have.length', 1);
+      cy.visit(`/reset-password/fake_token`);
+      // there seems to be a race condition on the change password page
+      // adding a small wait fixes the issue
+      cy.wait(300);
+      cy.get('button[type="submit"]').click();
+      cy.get('input[name="password"]:invalid').should('have.length', 1);
     });
   });
 
   context('Valid password entered', () => {
     it('shows password change success screen, with a default redirect button.', () => {
       cy.mockNext(200);
       cy.mockNext(200, fakeSuccessResponse);
-      page.goto(fakeToken);
-
-      page.typePasswordChange(
-        'thisisalongandunbreachedpassword',
-        'thisisalongandunbreachedpassword',
-      );
-      cy.contains(ChangePasswordPage.CONTENT.PASSWORDS_MATCH);
-      page.clickPasswordChange();
+      cy.visit(`/reset-password/fake_token`);
 
-      cy.contains(ChangePasswordPage.CONTENT.PASSWORD_CHANGE_SUCCESS_TITLE);
-      cy.contains(ChangePasswordPage.CONTENT.CONTINUE_BUTTON_TEXT).should(
+      cy.get('input[name="password"]').type('thisisalongandunbreachedpassword');
+      // there seems to be a race condition on the change password page
+      // adding a small wait fixes the issue
+      cy.wait(600);
```

Comment:
some waits are longer than others. Is this necessary?

Reply:
Mayyybe 🤷  - I suspect this wait is needed as there's a race around how long the api call to haveibeenpawnd in which case you're right, it should probably be the same. I'll update that.

Going forward we should keep an eye on this and adjust as required.

Reply:
I've fixed this in a later PR by using `intercept()` to make the wait actually _wait_ for the call